src/Features/CSharp/Portable/Formatting/CSharpFormattingInteractionService.cs

```diff
@@ -0,0 +1,463 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Indentation;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Indentation;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    [ExportLanguageService(typeof(IFormattingInteractionService), LanguageNames.CSharp), Shared]
+    internal partial class CSharpFormattingInteractionService : IFormattingInteractionService
+    {
+        internal const bool SupportsFormatDocumentConstant = true;
+        internal const bool SupportsFormatOnPasteConstant = true;
+        internal const bool SupportsFormatSelectionConstant = true;
+        internal const bool SupportsFormatOnReturnConstant = false;
```

Comment:
These constants were extracted for use in the editor layer version of this service. If I can remove that layer's implementation, these can go away.